In [56]:
import math
from math import factorial as fact
from scipy.special import comb

###### Question 12b:

Verify the following identity for $n = 1,2,3,4,5$

###### In Python

In [35]:
# combinatorial identities 
def fn1(n):
    result = 0
    for k in range(n):
        k+=1
        result += comb(n,k)*k**2
        
    return int(result)

fn2 = lambda n: int(2**(n-2)*(n)*(n+1))

In [46]:
for n in range(1,6):
    id1,id2 = fn1(n),fn2(n)
    if id1 == id2:
        print(f'for n={i}: combination 1: {id1} combination 2: {id2}')

for n=5: combination 1: 1 combination 2: 1
for n=5: combination 1: 6 combination 2: 6
for n=5: combination 1: 24 combination 2: 24
for n=5: combination 1: 80 combination 2: 80
for n=5: combination 1: 240 combination 2: 240


###### verification

a) for $n=1, k=1$

$
\begin{align}
 \sum_{k=1}^{n} {n \choose k} k^2 = 2^{n-2}n(n+1)
\end{align}
$

$
\begin{align}
  \implies {1 \choose 1} \cdot 1^2 = 2^{1-2}(1)(1+1)
\end{align}
$

$
\begin{align}
  \implies 1 = 1
\end{align}
$


b) for $n=2, k=1,2$

$
\begin{align}
  \implies {2 \choose 1} \cdot 1^2 + {2 \choose 2} \cdot 2^2 = 6
\end{align}
$

$
\begin{align}
  \implies 2^{2-2}(2)(2+1) = 6
\end{align}
$

$
\begin{align}
  \implies 6 = 6
\end{align}
$

c) for $n=3, k=1,2,3$

$
\begin{align}
  \implies {3 \choose 1} \cdot 1^2 + {3 \choose 2} \cdot 2^2 
   + {3 \choose 3} \cdot 3^2
   = 3^{3-2}(3)(3+1)
\end{align}
$

$
\begin{align}
  \implies 24 = 24
\end{align}
$

c) for $n=4, k=1,2,3,4$

$
\begin{align}
  \implies {4 \choose 1} \cdot 1^2 
   + {4 \choose 2} \cdot 2^2 
   + {4 \choose 3} \cdot 3^2
   + {4 \choose 4} \cdot 4^2
   = 4^{4-2}(4)(4+1)
\end{align}
$

$
\begin{align}
  \implies 80 = 80
\end{align}
$

c) for $n=5, k=1,2,3,4,5$

$
\begin{align}
  \implies {5 \choose 1} \cdot 1^2 
   + {5 \choose 2} \cdot 2^2 
   + {5 \choose 3} \cdot 3^2
   + {5 \choose 4} \cdot 4^2
   + {5 \choose 4} \cdot 5^2
   = 5^{5-2}(5)(5+1)
\end{align}
$

$
\begin{align}
  \implies 240 = 240
\end{align}
$

Combinatorial Proof:

k members (of the committee) can be selected from n people ${n \choose k}$ ways. One chairperson and one secretary can be selected from the committee (both can be the same $n \choose k$$k^2$ ways.

If the chair and the secretary are the same person then they can be $n 2^{n-1}$ because there are two possible states, in the group and out of the group and the exponent represents the total of the group less the chair and secretary. Finally the n value represents the total number of ways this can occur.

A chair person can be selected from a group of $n$ people $n \choose 1$ ways, and a secratary $n-1 \choose 1$ ways, which is the same as saying $n(n-1)$. 

Similar to above the remaining members can be selected $2^{n-2}$ ways. This results in $n(n-1)2^{n-2}$ for secretary and chair as different people.

Finally combining the two scenarios we get: $n 2^{n-1} + n(n-1)2^{n-2}$, which simplifies to $2^{n-2}(n+1)$


#### Chapter 2:

##### Question 8:

Suppose A and B are mutually exclusive events for which $P(A) = 0.3 \hspace{1mm} and 
\hspace{1mm} P(B) = 0.5$

1) Either A or B occur?
$P(A) + P(B) = 0.3 + 0.5 = 0.8$

2) A occurs but B does not: $P(A) = 0.3$

3) Both A and B occur:  $P(A and B) = 0$

##### Question 13:

Filler

###### Question 30:

a) Rebecca and Elise paired: <br>
-total choices for schools: $8 \choose 4$ and $9 \choose 4$ <br>
-after rebecca and elise are paired off: $7 \choose 3$ and $8 \choose 3$ <br>
-players can be paired: $4!$ and rebecca & elise: $3!$ <br>

$
\begin{align}
  P(R \cap E) = \dfrac{outcomes \hspace{1mm} where \hspace{1mm} 
  rebecca \hspace{1mm} and \hspace{1mm} elise \hspace{1mm} paired}{outcomes}
\end{align}
$

$
\begin{align}
  P(R \cap E) = \dfrac{{7 \choose 3} {8 \choose 3}3!}{{8 \choose 4} {9 \choose 4}4!}
  = \dfrac{1}{18}
\end{align}
$

###### calculations using python

In [54]:
total_choices = comb(8,4)*comb(9,4)
rebecca, elise = comb(7,3), comb(8,3)

In [61]:
# 0.05555 = 1/18

(math.factorial(3)*rebecca*elise)/(total_choices*math.factorial(4))

0.05555555555555555

a) Rebecca and Elise paired: <br>


$
\begin{align}
  P(R \cap E) = \dfrac{outcomes \hspace{1mm} where \hspace{1mm} 
  rebecca \hspace{1mm} and \hspace{1mm} elise \hspace{1mm} paired}{outcomes}
\end{align}
$

$
\begin{align}
  P(R \cap E) = \dfrac{{7 \choose 3} {8 \choose 3}3!}{{8 \choose 4} {9 \choose 4}4!}
  = \dfrac{1}{18}
\end{align}
$

##### Question T15

An urn contains $M$ white and $N$ black balls. If a random sample size of $r$ is chosen, what is the probability that it contains $k$ white balls?

-If $S$ is the sample space containing all possible choices or $r$, there are $M + N$ choices in $S$. <br>
-$S$ consists of ${M+N \choose r}$ total possibilities <br>
-Since every outcome is equally likely: $P(E) = \dfrac{E}{S}$ <br>
-Since $M$ represents combinations of white balls, the event $k$ white balls are drawn is represented by $N \choose k$ <br>
-Also $N$ represents combinations of black balls: $M \choose r-k $. $r$ is calculated for black balls by subtracting off $k$ balls chosen in previous statement <br>
-$E$ = ${M \choose k} {N \choose r-k}$ <br>
-Finally the probability of choosing $r$ balls from $M$ white and $N$ black is: <br>
$P(E) = \dfrac{{M \choose k} {N \choose r-k}}{{M+N \choose r}}$